NorthstarLauncher/main.cpp

```diff
@@ -223,14 +223,44 @@ void EnsureOriginStarted()
 	CloseHandle(pi.hThread);
 }
 
+[[nodiscard]] static std::optional<std::wstring> get_environment_variable(const wchar_t* var_name) noexcept
+{
+#ifdef _MSC_VER
+	wchar_t* buffer = nullptr;
+	size_t buffer_count = 0;
+	const auto error = _wdupenv_s(&buffer, &buffer_count, var_name);
+	if (!buffer || buffer_count == 0 || error)
+		return std::nullopt;
+
+	std::wstring str {buffer, buffer_count};
+	free(buffer);
+
+	return str;
+#elif defined(__MINGW32__)
+	// For some reason mingw doesn't have a definition of _wdupenv_s despite it being declared
+	std::vector<wchar_t> buffer;
+
+	while (true)
+	{
+		const auto size = GetEnvironmentVariableW(var_name, buffer.data(), buffer.size());
+		if (size == 0)
+			return std::nullopt;
+
+		if (size < buffer.size())
+			return std::wstring {buffer.data()};
+
+		buffer.resize(size);
+	}
+#endif
+}
```

Comment:
This is not at all portable. We should really be striving for purely c++20 compatible code here, and this is not that.

Reply:
I assumed that not unnecessarily changing the behaviour of the MSVC code would be more desirable (afaict `_wdupenv_s` is better since you don't have to call it multiple times to get the environment variable). If you'd like to change to `getenv` for both I can do that.